### PR TITLE
Feature/6111 index and search source information

### PR DESCRIPTION
**Pull request by melsawy**

### CODE REVIEW

File: app/models/project_source.rb [hunk 1]
```diff
@@ -14,6 +14,8 @@ class ProjectSource < ActiveRecord::Base
   validates :source_id, uniqueness: { scope: :project_id }
   before_validation :set_account, on: :create
 
+  after_create :add_elasticsearch_data
```
Comment (caiosba) [inline app/models/project_source.rb]: Sawy, is there a problem if we run this in background?

File: lib/check_search.rb [hunk 2]
```diff
@@ -77,8 +74,9 @@ def should_hit_elasticsearch?
     !(@options['status'].blank? && @options['tags'].blank? && @options['keyword'].blank?)
   end
 
-  def medias_build_search_query
+  def medias_build_search_query(assocated_type = 'ProjectMedia')
```
Comment (caiosba) [inline lib/check_search.rb]: Typo: `assocated` should be `associated` :)

**Comment by caiosba** [comment]
I just left two comments, Sawy... other than that looks good, thanks for the complete tests. Please also check the coverage and the Code Climate issue above.

File: app/models/project_source.rb (hunk 1 above)
Comment (melsawy) [inline app/models/project_source.rb]: Yes Caio always need to set parent document directory as we have an issue when create parent/child on background - we have a mantis ticket with this issue

File: lib/check_search.rb (hunk 2 above)
Comment (melsawy) [inline lib/check_search.rb]: will fix it

File: app/models/project_source.rb (hunk 1 above)
Comment (caiosba) [inline app/models/project_source.rb]: Ok thanks

**Pull request merged**